rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceConfigurationParser.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.nodesource;
+
+import java.util.HashMap;
+
+import org.ow2.proactive_grid_cloud_portal.rm.client.NodeSourceConfiguration;
+import org.ow2.proactive_grid_cloud_portal.rm.client.PluginDescriptor;
+import org.ow2.proactive_grid_cloud_portal.rm.client.RMController;
+
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONException;
+import com.google.gwt.json.client.JSONObject;
+
+
+public class NodeSourceConfigurationParser {
+
+    private RMController controller;
+
+    public NodeSourceConfigurationParser(RMController controller) {
+        this.controller = controller;
+    }
+
+    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) throws ImportException {
+
+        JSONObject jsonObject;
+
+        try {
+            jsonObject = this.controller.parseJSON(json).isObject();
+        } catch (JSONException e) {
+            throw new ImportException("The imported node source is not a valid JSON file", e);
+        }
+
+        try {
+            String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
+            boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
+
+            JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+            String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName")
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "pluginName" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)